server/blob/blob-sharing-api/src/main/java/org/apache/james/blob/sharing/api/BlobSharingMechanism.java

```diff
@@ -0,0 +1,38 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.sharing.api;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.core.MailAddress;
+import org.reactivestreams.Publisher;
+
+public interface BlobSharingMechanism {
+    @FunctionalInterface
+    interface ShareeStage {
+        ExplanationStage with(MailAddress sharee);
+    }
+
+    @FunctionalInterface
+    interface ExplanationStage {
+        Publisher<Void> withExplanation(String explanation);
```

Comment:
This is always gonna be wrapped by a task so I just think using reactor in this context adds dificulty